BackendWebApp/src/main/java/uk/ac/ebi/biostd/webapp/application/rest/service/PendingSubmissionUtil.java

```diff
@@ -0,0 +1,128 @@
+package uk.ac.ebi.biostd.webapp.application.rest.service;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.ac.ebi.biostd.webapp.application.rest.dto.PendingSubmissionDto;
+import uk.ac.ebi.biostd.webapp.application.rest.dto.PendingSubmissionListItemDto;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+@Component
+@Slf4j
+@AllArgsConstructor
+public class PendingSubmissionUtil {
+
+    private static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+
+    private static final String ACCNO_PREFIX = "TMP_";
+
+    private final ObjectMapper objectMapper;
+
+    public String asString(PendingSubmissionDto dto) {
+        return objectMapper.valueToTree(dto).toString();
+    }
+
+    public Optional<PendingSubmissionDto> parse(String data) {
+        try {
+            return Optional.of(objectMapper.readValue(data, PendingSubmissionDto.class));
+        } catch (IOException e) {
+            log.error("error while parsing pending submission", e);
+        }
+        return Optional.empty();
+    }
+
+    public PendingSubmissionListItemDto convert(PendingSubmissionDto pendingSubmission) {
+        JsonNode node = pendingSubmission.getData();
+        return PendingSubmissionListItemDto.builder()
+                .accno(pendingSubmission.getAccno())
+                .mtime(pendingSubmission.getModificationTimeInSeconds())
+                .rtime(getReleaseTimeInSeconds(node).orElse(null))
+                .title(getTitle(node).orElse(""))
+                .build();
+    }
+
+    public Optional<PendingSubmissionDto> create(String pageTab) {
```

Comment:
lests use lombok @Sneakythrows to avoid code to handle this exception that we actually we only log error. Please ping me for details about this annotation

Reply:
Not sure I can use it here. As the idea is to return Optional.empty in case of error.